src/modules/swap/containers/SwapWidget/index.tsx

```diff
@@ -65,7 +67,7 @@ export function SwapWidget() {
   const { independentField, recipient } = swapState
   const showRecipientControls = useShowRecipientControls(recipient)
   const isEthFlow = useIsEthFlow()
-  const shouldZeroApprove = useShouldZeroApproveSwap()
+  const shouldZeroApprove = useShouldZeroApprove(sellAmountMaxSpend)
```

Comment:
@tukantje please, pay attention here

Reply:
Yes, the goal of these changes is the unification of getting an amount to approve - `slippageAdjustedSellAmount`.
Now this value is encapsulated in `TradeDerivedState` (each trade widget implements this interface).
Also, `<ZeroApprovalModal />` is integrated in `<TradeWidgetModals/>` which in turn provides the zero approval modal for each trade widget out of the box.